pennylane/ops/op_math/condition.py

```diff
@@ -102,7 +103,111 @@ def adjoint(self):
         return Conditional(self.meas_val, self.base.adjoint())
 
 
-def cond(condition, true_fn: Callable, false_fn: Optional[Callable] = None, elifs=()):
+class CondCallable:  # pylint:disable=too-few-public-methods
+    """Base class to represent a conditional function with boolean predicates.
+
+    Args:
+        condition (bool): a conditional expression
+        true_fn (callable): The function to apply if ``condition`` is ``True``
+        false_fn (callable): The function to apply if ``condition`` is ``False``
+        elifs (List(Tuple(bool, callable))): A list of (bool, elif_fn) clauses.

```

Comment:
What do you think about the type hint including `MeasurementValue`? Honestly, I'm fine either way since I feel like it's still ambiguous how user facing `MeasurementValue` really is.
```suggestion
        condition (Union[bool, pennylane.measurements.MeasurementValue]): a conditional expression
        true_fn (callable): The function to apply if ``condition`` is ``True``
        false_fn (callable): The function to apply if ``condition`` is ``False``
        elifs (List(Tuple(Union[bool, pennylane.measurements.MeasurementValue], callable))): A list of (bool, elif_fn) clauses.

```

Reply:
I didn't think elifs worked with measurement values?

Reply:
Ah I misread this when I first looked! I thought that `CondCallable` does not accept `MeasurementValue`'s, based on this part of the logic:

```python
if not isinstance(condition, MeasurementValue):
    # The condition is not a mid-circuit measurement.
    if true_fn is None:
        return lambda fn: CondCallable(condition, fn)

    return CondCallable(condition, true_fn, false_fn, elifs)
```

So I'll leave this as-is for the time being